src/main.ts

```diff
@@ -138,10 +139,34 @@ function createIdGenerator(): () => Id {
   }
 }
 
-function inferTSConfig(projectPath: string): boolean {
-  // TODO: Move logic from https://github.com/sourcegraph/sourcegraph/pull/30091
-  // into this function.
-  return false;
+/** Returns the list of directory paths under dir which contain a file
+  * with name matchingName.
+*/
+function listDirsRecursive(dir: string, matchingName: string): string[] {
+  let paths: string[] = [];
+  var basenames = fs.readdirSync(dir);
+  for (const basename of basenames) {
+    const path = dir + '/' + basename;
+    const stat = fs.statSync(path);
+    if (stat.isDirectory()) {
+      paths = paths.concat(listDirsRecursive(path, matchingName));
+    } else if (stat.isFile() && basename == matchingName) {

```

Comment:
nit: i think there's a linter that catches these
```suggestion
    } else if (stat.isFile() && basename === matchingName) {

```

Reply:
Just ran `npm run eslint` and I got this. 😢

```
✖ 575 problems (471 errors, 104 warnings)
  372 errors and 2 warnings potentially fixable with the `--fix` option.
```